oz/aws_cdn/__init__.py

```diff
@@ -45,7 +45,10 @@ def get_bucket(s3_bucket=None, validate=False):
     if S3Connection != None:
         settings = oz.settings
         s3_bucket = s3_bucket or settings["s3_bucket"]
-        return S3Connection(settings["aws_access_key"], settings["aws_secret_key"]).get_bucket(s3_bucket, validate=validate)
+        opts = {}
+        if settings["s3_host"]:
+            opts["host"] = settings["s3_host"]
+        return S3Connection(settings["aws_access_key"], settings["aws_secret_key"], **opts).get_bucket(s3_bucket, validate=validate)
```

Comment:
This could be simplified:

```
conn = S3Connection(settings["aws_access_key"], settings["aws_secret_key"], host=settings.get("s3_host"))
return conn.get_bucket(s3_bucket, validate=validate)
```

Reply:
The way boto handles 'host' disallows this because it uses a special `NoHostProvided` class to determine if it needs to look for a host via some other means, like boto's config. If I set `host` to `None` (or anything other than `NoHostProvided`, actually), you will get unexpected results.